### DISCUSSION TITLE
Is there a way to add all .md files without adding them one by one in index.rst toctree for sphinx?

### DISCUSSION THREAD

**Discussion by monajalal** [executablebooks/MyST-Parser#943] [General]
I am currently adding the .md files one by one in index.rst under toctree so that they would be rendered in the html. Is there a way to add them all without listing them like so?

**Comment by chrisjsewell** [comment]
Heya yeh toctree has a glob option https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-toctree

**Comment by monajalal** [comment]
@chrisjsewell 

Thanks a lot for your response.

When I tried

.. toctree::
   :caption: Contents
   *.md

It didn't render any of my .md files in current dir. Also, I cannot do it for .md files in documentation folder

.. toctree::
   :caption: Contents
   documentation/*.md


However,

.. toctree::
   :caption: Contents
   README.md
   myfile.md

works.

Could you please suggest how to fix this?

**Reply by chrisjsewell** [reply]
You have to add the `:glob:` option 😄

**Reply by monajalal** [reply]
@chrisjsewell Thank you.

I tried both of these none renedered any of the .md files in the current directory.

.. toctree::
   :glob:
   ./*.md

or

.. toctree::
   :glob:
   *.md


Also tried this

.. toctree::
   :glob:
   source/*.md


my md files are inside source folder but I run make html in documentation_tools folder

(base) mona@DOS:~/doc_project/documentation_tools/source (main)$ ls *.md
-rw-rw-r-- 1 mona mona   33 Jul 10 15:03 myfile.md
-rw-rw-r-- 1 mona mona 1.3K Jul 10 15:37 README.md


(base) mona@DOS:~/doc_project/documentation_tools (main)$ make html


![Screenshot from 2024-07-12 12-02-02](https://github.com/user-attachments/assets/2e211b70-d2f8-41be-93e0-d5fa1ecb0c7a)

**Reply by monajalal** [reply]
@chrisjsewell I was wondering if you may know the answer to this?

**Reply by chrisjsewell** [reply]
For any restructuredtext directive you need a space between the options and the content, and you also don't use the extension in `toctree`

```
.. toctree::
	:glob:

	*
```

**Reply by monajalal** [reply]
@chrisjsewell hi Chris 

I tried aligning :glob: right under `e` from toctree and that didn't work.
I also add a line of space between :glob: and *.md and that didn't work.

![image](https://github.com/user-attachments/assets/68504b5a-f13b-455d-a94f-caf473bc41d4)



Please note the Content indendation is correct and works.

I have .md files both where I run the code (documentation_tools folder) (where I run make html) and also md files in source dir (documentation_tools/source), none is showing up using glob directive. 

I also used what you wrote above as well (just the `*`) that didn't work too.

Do you know what am I missing?

**Comment by 72757373656c6c** [comment]
Are you getting any error messages?

Check the value of `source_suffix` in `conf.py`. 

```
source_suffix = {
    ".md": "markdown",
    ".rst": "restructuredtext",
}
```

If there isn't a reason that you require the index file to be in rst format, it can also be in markdown. 

index.md
```
# Welcome

:::{toctree}
:glob:

*/index
[a-z]*
:::
```

index.rst
```
# Welcome

.. toctree::
   :glob:

   */index
   [a-z]*
```

https://myst-parser.readthedocs.io/en/latest/syntax/organising_content.html#using-toctree-to-include-other-documents-as-children

- Sphinx 8.0.2
- myst-parser 4.0.0
